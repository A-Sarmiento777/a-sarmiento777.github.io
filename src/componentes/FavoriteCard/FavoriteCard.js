import React from 'react';

const FavoriteCard = ({ removeItemFromCartFav, producto }) => {
    return (
        <tr key="index">
            <td><img src={process.env.PUBLIC_URL + `/Assets/${producto.type}/${producto.name}.jpg`} width='100' alt="logos" className="img-fluid py-2" /></td>
            <td>{producto.name}</td>
            <td>{producto.price}</td>
            <td onClick={() => removeItemFromCartFav(producto.id)} style={{ color: 'red' }}>Remove</td>
        </tr>
    );
}

export default FavoriteCard;