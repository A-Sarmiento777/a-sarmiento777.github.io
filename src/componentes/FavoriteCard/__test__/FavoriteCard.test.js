// Link.react.test.js
import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import FavoriteCard from "../FavoriteCard";

test("FavoriteCard Card test", () => {
  const removeItemFromCartFav = jest.fn(() => 1);

  let testRenderer = renderer.create(
    <FavoriteCard
      priceFormatter={() => {}}
    removeItemFromCartFav={removeItemFromCartFav} 
    producto={{ id: 1, type: 'saadad', name: "Test", price: "6.7" }}
    ></FavoriteCard>
  );

  let tree = testRenderer.toJSON();
  expect(tree).toMatchSnapshot();

  tree.children[3].props.onClick();

  expect(removeItemFromCartFav).toHaveBeenCalled();
  expect(removeItemFromCartFav).toBeCalledTimes(1);

//   testRenderer.update(
//     <Router>
//       <FavoriteCard
//     removeItemFromCartFav={removeItemFromCartFav} 
//     producto={{ id: 1, name: "Test", price: "6.7" }}
//     ></FavoriteCard>
//     </Router>
//   );

//   tree = testRenderer.toJSON();
//   expect(tree).toMatchSnapshot();

});
