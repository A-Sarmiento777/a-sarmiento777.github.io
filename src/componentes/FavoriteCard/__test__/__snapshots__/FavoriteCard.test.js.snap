// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteCard Card test 1`] = `
<tr>
  <td>
    <img
      alt="logos"
      className="img-fluid py-2"
      src="/Assets/saadad/Test.jpg"
      width="100"
    />
  </td>
  <td>
    Test
  </td>
  <td />
  <td
    onClick={[Function]}
    style={
      Object {
        "color": "red",
      }
    }
  >
    Remove
  </td>
</tr>
`;
